spec/controllers/api/v1/interactive_pages_controller_spec.rb

```diff
@@ -136,4 +136,109 @@ def add_interactive_to_section(page, interactive, section)
+        }.to_json)
+      end
+    end
+
+    describe "in a private activity" do
+      let (:publication_status) { "private" }
+
+      it "returns an error" do
+        xhr :post, "set_linked_interactives", {id: page.id}
+        expect(response.status).to eq(200)
+        expect(response.content_type).to eq("application/json")
+        expect(response.body).to eql({
+          success: false,
+          message: "You are not authorized to set linked interactives for the requested page"
+        }.to_json)
+      end
+    end
+
+    describe "on a page with interactives" do
+      before :each do
+        # need to be an author to create linked interactives
+        sign_in author
+
+        add_interactive_to_section(page, interactive1, InteractivePage::INTERACTIVE_BOX)
+        add_interactive_to_section(page, interactive2, InteractivePage::HEADER_BLOCK)
+        add_interactive_to_section(page, interactive3, nil) # nil is assessment block
+        add_interactive_to_section(page, interactive4, InteractivePage::INTERACTIVE_BOX)
+        add_interactive_to_section(page, interactive5, InteractivePage::HEADER_BLOCK)
+        page.save!(validate: true)
+        page.reload
+      end
+
+      describe "returns an error when there is no source param" do
+        it "returns an error" do
+          xhr :post, "set_linked_interactives", {id: page.id}
+          expect(response.status).to eq(200)
+          expect(response.content_type).to eq("application/json")
+          expect(response.body).to eql({
+            success: false,
+            message: "Missing sourceId parameter in request"
+          }.to_json)
+        end
+      end
+
+      describe "returns an error when there is an invalid source param" do
+        it "returns an error" do
+          xhr :post, "set_linked_interactives", {
+            id: page.id,
+            sourceId: 1000000000
+          }
+          expect(response.status).to eq(200)
+          expect(response.content_type).to eq("application/json")
+          expect(response.body).to eql({
+            success: false,
+            message: "Invalid sourceId parameter in request"
+          }.to_json)
+        end
+      end
+
+      it "allows a 1:1 link" do
+        xhr :post, "set_linked_interactives", {
+          id: page.id,
+          sourceId: interactive1.page_item.id,
+          linkedInteractives: {
+            "#{interactive2.page_item.id}": {label: "two"}
+          },
+          linkedState: interactive3.page_item.id
+        }
+        expect(response.status).to eq(200)
+        expect(response.content_type).to eq("application/json")
+        expect(response.body).to eql({
+          success: true
+        }.to_json)
+      end
+
+      it "allows a 1:N link" do
+        xhr :post, "set_linked_interactives", {
+          id: page.id,
+          sourceId: interactive1.page_item.id,
+          linkedInteractives: {
+            "#{interactive2.page_item.id}": {label: "two"},
```

Comment:
I just realized seeing this spec, that really it would be better if a source interactive could use the same target in multiple links. I think of the labels like slots or form fields. So it seems probable that some use case will come up where an author should should be able to fill in multiple form fields with the same interactive. 

So then it makes more sense for the structure to be:
```
{ "two": {id: "#{interactive2.page_item.id}"},
  "three": {id: "#{interactive3.page_item.id}"}
}
```
That still restricts the interactive to simple maps for referencing interactives. It doesn't make it easy to do lists or arrays. But an interactive could do `array_1`, `array_2`, ... if it needs to make an array.

I realize this would require lots of changes to this PR, but I think it is worth doing now before it becomes harder later.

Reply:
Sounds like it should just be an array then.  I also realized this morning we don't have a way to delete existing links so I thought of extending the existing type to allow for that if value is null (like how Firebase does deletes).  If we use an array the item type could be `{id: number, label: string} || {id: number, delete: true}` to enable deletes.

Reply:
I thought based on the code in this PR that it was deleting all existing links with this source. It looked like it did that delete, then recreates them each time. That seems more simple to me instead of doing explicit deletes. We could improve its efficiency later. But the basic idea is that the interactive needs to send all of its links each time. If it doesn't include a previous link, then it gets deleted.

An array sounds good. If we preserve the order of the array, then interactives can support maps, and labeled arrays. 
Something like:

```
[
 {id: "interactive_123", label: "one"},
 {id: "interactive_234", label: "two"},
 {id: "interactive_345", label: "two"}
]
```

So in that example "one" is a simple map to a single interactive. And "two" is an array of 2 interactives. 

To preserve the order of the array we'd need to add a position field to the LinkedPageItems.

Reply:
oh also I used an string for the id that gives us the most flexibility in the future.